weave

```diff
@@ -734,17 +740,47 @@ ipam_claim() {
     done
 }
 
-# If we don't already have any CIDRs passed from the command-line, get
-# one from IPAM. $1 is the full container id.
-get_ipam_cidr() {
-    if [ -n "$CIDR_ARGS" ] ; then
+# Call IPAM as necessary to allocate addresses
+# $1 is the full container id and following args are previously parsed CIDR_ARGS.
+# Returns ALL_CIDRS and IPAM_CIDRS
+ipam_cidrs() {
+    CONTAINER_ID="$1"
+    shift 1
+    if [ $# -gt 0 ] ; then
```

Comment:
I suggest inverting this, putting the (shorter) 0-arg code into the `then` block.